buildkitd.go

```diff
@@ -74,7 +74,7 @@ func SpawnBuildkitd(req Request, opts *BuildkitdOpts) (*Buildkitd, error) {
 	if os.Getuid() == 0 {
 		cmd = exec.Command("buildkitd", buildkitdFlags...)
 	} else {
-		cmd = exec.Command("rootlesskit", append([]string{"buildkitd"}, buildkitdFlags...)...)
+		cmd = exec.Command("rootlesskit", append([]string{"buildkitd", "--rootless"}, buildkitdFlags...)...)
```

Comment:
Is this one related or is it a bug left from before?

Reply:
Not strictly NixOS-specific, and certainly reasonable to separate out into a different PR to be individually evaluated. While debugging, it came up that buildkit (or at least, the version shipped in NixOS 21.11) has a `--rootless` which is documented to enable configuration changes for better compatibility with rootlesskit -- but that said, I didn't go back after getting other issues ironed out to confirm that we were unable to run tests without it.

I'll do that now and follow up with findings.